WindowManager/app/src/main/java/com/example/windowmanagersample/embedding/SplitActivityBase.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.windowmanagersample.embedding;
+
+import static android.app.PendingIntent.FLAG_IMMUTABLE;
+
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.util.LayoutDirection;
+import android.util.Log;
+import android.util.TypedValue;
+import android.view.View;
+import android.widget.CompoundButton;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.util.Consumer;
+import androidx.window.core.ExperimentalWindowApi;
+import androidx.window.embedding.ActivityFilter;
+import androidx.window.embedding.ActivityRule;
+import androidx.window.embedding.EmbeddingRule;
+import androidx.window.embedding.SplitController;
+import androidx.window.embedding.SplitInfo;
+import androidx.window.embedding.SplitPairFilter;
+import androidx.window.embedding.SplitPairRule;
+import androidx.window.embedding.SplitPlaceholderRule;
+import com.example.windowmanagersample.databinding.ActivitySplitActivityLayoutBinding;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import kotlin.OptIn;
+
+/**
+ * Sample showcase of split activity rules. Allows the user to select some split configuration
+ * options with checkboxes and launch activities with those options applied.
+ */
+@OptIn(markerClass = ExperimentalWindowApi.class)
+public class SplitActivityBase extends AppCompatActivity
```

Comment:
Just curious if there is a reason this `Activity` is written in `Java` instead of `Kotlin`.

Reply:
This was originally in Java. I can look into this in the future to see if there is a better way to consume the events then a `SplitInfoCallback`.